Modix.Bot/Behaviors/PromotionPollBehavior/PromotionDialogStartupBehavior.cs

```diff
@@ -0,0 +1,58 @@
+using System;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Caching.Memory;
+using Microsoft.Extensions.DependencyInjection;
+using Modix.Data.Repositories;
+using Modix.Services;
+
+namespace Modix.Bot.Behaviors
+{
+    public class PromotionDialogStartupBehavior :IBehavior
+    {
+
+        private IMemoryCache _memoryCache { get; }
+
+        private IServiceProvider _serviceProvider { get; }
+
+        private IPromotionDialogRepository _promotionDialogRepository { get; set; }
+
+        public PromotionDialogStartupBehavior(IMemoryCache memoryCache,
+            IServiceProvider serviceProvider)
+        {
+            _memoryCache = memoryCache;
+            _serviceProvider = serviceProvider;
+        }
+
+        public async Task StartAsync()
+        {
+            using var serviceScope = _serviceProvider.CreateScope();
+            _promotionDialogRepository = serviceScope.ServiceProvider.GetRequiredService<IPromotionDialogRepository>();
```

Comment:
BIG RED FLAG!

You're saving this instance outside of the scope of the `using` statement that will dispose it.